Simplenote/Classes/SPAuthViewController.swift

```diff
@@ -628,12 +635,59 @@ extension SPAuthViewController: SPTextInputViewDelegate {
     }
 }
 
+// MARK: - Passkeys
+//
 extension SPAuthViewController: ASAuthorizationControllerPresentationContextProviding {
     func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
         view.window!
     }
 }
 
+extension SPAuthViewController: ASAuthorizationControllerDelegate {
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: any Error) {
+        passkeyAuthFailed(error)
+        print(error.localizedDescription)
+    }
+
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+
+        switch authorization.credential {
+
+        case let credential as ASAuthorizationPlatformPublicKeyCredentialAssertion:
+            let response = PasskeyAuthResponse(from: credential)
+
+            performPasskeyAuthentication(with: response)
+        default:
+            break
+        }
+    }
+
+    private func performPasskeyAuthentication(with response: PasskeyAuthResponse) {
+        Task { @MainActor in
+            guard let json = try? JSONEncoder().encode(response),
+                  let response = try? await PasskeyRemote().verifyPasskeyLogin(with: json),
+                  let verifyResponse = try? JSONDecoder().decode(PasskeyVerifyResponse.self, from: response) else {
+                passkeyAuthFailed(PasskeyError.authFailed)
+                return
+            }
+
+            controller.simperiumService.authenticate(withUsername: verifyResponse.username, token: verifyResponse.accessToken)
+            unlockInterface()
```

Comment:
**Ideally** the ViewController doesn't need to deal with json encoding / decoding. More on that below!!